### PR TITLE
Adds test that affects the proposer selection

### PR DESCRIPTION
**Pull request by GabrielAstieres**
This aims to add a test where for the edge case where something happening in epoch N leads to a change in the next-epoch proposer schedule.

### CODE REVIEW

**Comment by mkalinin** [comment]
Thanks for this test! I think we should also have the same for pure exit case, consolidation affects proposer selection because proposer is exiting and its balance becomes zero

File: tests/core/pyspec/eth2spec/test/phase0/epoch_processing/test_process_effective_balance_updates.py [hunk 1]
```diff
@@ -4,14 +4,37 @@
     set_compounding_withdrawal_credential,
 )
 from eth2spec.test.helpers.forks import is_post_electra
-
+from eth2spec.test.helpers.state import (next_epoch)
```
Comment (mkalinin) [inline tests/core/pyspec/eth2spec/test/phase0/epoch_processing/test_process_effective_balance_updates.py:7]:
```suggestion
from eth2spec.test.helpers.state import next_epoch
```

File: tests/core/pyspec/eth2spec/test/electra/epoch_processing/test_process_pending_consolidations.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 )
 from eth2spec.test.context import (
     spec_state_test,
-    with_electra_and_later,
+    with_electra_and_later
```
Comment (mkalinin) [inline tests/core/pyspec/eth2spec/test/electra/epoch_processing/test_process_pending_consolidations.py:7]:
```suggestion
    with_electra_and_later,
```
usual thing in python to leave commas at the end, should we add one more line this trick would reduce the size of the diff

File: tests/core/pyspec/eth2spec/test/phase0/epoch_processing/test_process_effective_balance_updates.py [hunk 3]
```diff
@@ -4,14 +4,37 @@
...
+
+    yield 'pre', state
+
+    future_state = state.copy()
+    next_epoch(spec, future_state)
+    next_epoch(spec, future_state)
+    proposer_next_epoch_after_change = spec.get_beacon_proposer_index(future_state)
+
+    assert proposer_next_epoch != proposer_next_epoch_after_change
+    yield 'post', state
```
Comment (mkalinin) [inline tests/core/pyspec/eth2spec/test/phase0/epoch_processing/test_process_effective_balance_updates.py:35]: For this test the client will do the following:
1) load `pre` state
2) apply `process_effective_balance_updates` to it
3) check if the result matches `post` state

And effectively the test will fail on clients side. If you want to do epoch transition between `pre` and `post` state then you should rather use `sanity/test_slots` like it is done here: https://github.com/ethereum/consensus-specs/blob/dev/tests/core/pyspec/eth2spec/test/electra/sanity/test_slots.py#L9 — note that this test yields pre-state right before the epoch transition to minimize a number of slots to run through on a client side.

why should we move two epochs into the future?
what if you do `state.balance[proposer_next_epoch] = 0` and then run the epoch processing which will set effective_balance to 0 does the desired thing?

Comment (GabrielAstieres) [reply tests/core/pyspec/eth2spec/test/phase0/epoch_processing/test_process_effective_balance_updates.py]: Hey @mkalinin thank you **SO** much for your help. Your explanation makes a lot of sense and allowed me to understand what I was dealing with. 


I moved this to `sanity/test_slots` and using the helper function to do the epoch processing. Only the second time however, because I want it to generate the artifacts in without duplicates. Let me know if this correct, but with this, the test is passing.

Additionally, I created this test in the Electra folder, but I guess this could go in `phase0`, do you have a preference there?

**Comment by GabrielAstieres** [comment]
I the end, based on recommendations, I did move the test to `phase0` so that it applies to all the fork.

**Review by mkalinin — APPROVED** [review]
Looks good! 👍
